sdm/src/main/java/com/sap/cds/sdm/service/SDMServiceImpl.java

```diff
@@ -27,92 +36,86 @@ public JSONObject createDocument(
       throws IOException {
     String accessToken;
     Map<String, String> finalResponse = new HashMap<>();
-
-    OkHttpClient client = new OkHttpClient();
     accessToken = TokenHandler.getDIToken(jwtToken, sdmCredentials);
 
     String sdmUrl = sdmCredentials.getUrl() + "browser/" + cmisDocument.getRepositoryId() + "/root";
 
-    try {
-      byte[] fileContent = IOUtils.toByteArray(cmisDocument.getContent());
-      RequestBody fileBody =
-          RequestBody.create(fileContent, MediaType.parse("application/octet-stream"));
-
-      RequestBody requestBody =
-          new MultipartBody.Builder()
-              .setType(MultipartBody.FORM)
-              .addFormDataPart("cmisaction", "createDocument")
-              .addFormDataPart(
-                  "objectId", cmisDocument.getFolderId()) // Note: removed quotes from folderId
-              .addFormDataPart("propertyId[0]", "cmis:name")
-              .addFormDataPart("propertyValue[0]", cmisDocument.getFileName())
-              .addFormDataPart("propertyId[1]", "cmis:objectTypeId")
-              .addFormDataPart("propertyValue[1]", "cmis:document")
-              .addFormDataPart("succinct", "true")
-              .addFormDataPart("filename", cmisDocument.getFileName(), fileBody)
-              .build();
-
-      handleDocumentCreationRequest(
-          cmisDocument, client, requestBody, sdmUrl, accessToken, finalResponse);
-
+    try (CloseableHttpClient httpClient =
+        HttpClients.custom()
+            .setDefaultRequestConfig(
+                RequestConfig.custom()
+                    .setConnectTimeout(SDMConstants.TIMEOUT * 1000)
+                    .setSocketTimeout(SDMConstants.TIMEOUT * 1000)
+                    .setConnectionRequestTimeout(SDMConstants.TIMEOUT * 1000)
+                    .build())
+            .build()) {
+
+      HttpPost uploadFile = new HttpPost(sdmUrl);
+      MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+      uploadFile.setHeader("Authorization", "Bearer " + accessToken);
+      // Add file to the form
+      builder.addBinaryBody(
```

Comment:
Can we try this approach? Even if the performance are comparable using inputStream will have edge over binary body when we deal with multiple upload requests in parallel 
```
InputStreamBody inputStreamBody = new InputStreamBody(
                    contentStream,
                    ContentType.create(cmisDocument.getMimeType()),
                    cmisDocument.getFileName()
            );
```

Reply:
added

Reply:
This will not work as internally it chunks the files and every chunk it is calling the createAttachmentHandler leading to duplicate issue

Reply:
Let us use the cap sdk library in the next pr instead of using httpclient. Fine for this PR